### PR TITLE
[SC] Added response handler for invalid txs

### PR DESCRIPTION
**Pull request by hyunsooda**
## Proposed changes

An observation that some bridge txs continuously stay in the bridge tx pool was reported. In order to remove bridge tx from the pool, the corresponding receipt from the counterpart chain must arrive as an answer of the successful process. That txs are resent until that corresponding receipt arrives. It makes sense. **However, the problem is the sync is not matched between the sender side and receiver side.** 

As a pipeline of the current implementation, the sender sends two RPC requests: 
 - sending bridge tx to be mined 
 - receiving back the corresponding receipt at every newly generated block (i.e., every 1s)
 
The receiver receives two requests and handles, but that tx is not mined in the new block at the time of the request, which causes no receipt response at this point. After that, the sender sends two requests again. The number of tx is two since he didn't get a receipt from the previous request. The receiver's new block now contains the previous tx and sends a receipt but doesn't include the fresh tx that comes in this request. The same scenario is looped and pending tx never becomes zero.

This PR adds:
- A request and response handler for invalid bridge txs sends invalid bridge tx hashes back to the sender so that the sender can remove those txs. An example of an invalid tx is "already known tx". (There's no error code for the type of invalid txs) 
- Do not send anchor tx if the anchor period does not arrive.

Refer to optimization issue at #1294.  

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: node/sc/protocol.go [hunk 1]
```diff
@@ -38,10 +38,11 @@ const (
...
 	ServiceChainParentChainInfoResponseMsg = 0x04
 	ServiceChainParentChainInfoRequestMsg  = 0x05
+	ServiceChainInvalidTxResponseMsg       = 0x06
 
-	ServiceChainCall     = 0x06
-	ServiceChainResponse = 0x07
-	ServiceChainNotify   = 0x08
+	ServiceChainCall     = 0x07
+	ServiceChainResponse = 0x08
+	ServiceChainNotify   = 0x09
```
Comment (aidan-kwon) [inline node/sc/protocol.go:45]: This kind of protocol update can make a mismatch in the message protocol.
Why don't you increase the protocol version? We can consider a better protocol version management policy also

File: node/sc/sub_bridge_handler.go [hunk 2]
```diff
@@ -317,17 +322,47 @@ func (sbh *SubBridgeHandler) LocalChainHeadEvent(block *types.Block) {
...
+	var txHashes []common.Hash
+	if err := msg.Decode(&txHashes); err != nil && err != rlp.EOL {
+		return errResp(ErrDecode, "msg %v: %v", msg, err)
+	}
+	txPool := sbh.subbridge.GetBridgeTxPool()
+	for _, txHash := range txHashes {
+		if tx := txPool.Get(txHash); tx != nil {
+			if err := txPool.RemoveTx(tx); err != nil {
+				logger.Trace("Failed to remove bridge tx",
+					"txType", tx.Type(), "txNonce", tx.Nonce(), "txHash", tx.Hash())
```
[the remark this answers is not in the document]
Comment (hyunsooda) [reply node/sc/sub_bridge_handler.go]: Thank you—changed log level with the error. Let me ask, what do you mean print the error message? `stdout`?

File: node/sc/protocol.go (hunk 1 above)
Comment (hyunsooda) [reply node/sc/protocol.go:45]: What a great catch! Thank you so much. The SC protocol version scope is currently going on another PR(#1248). Once that PR is finalized, let me update it.

File: node/sc/sub_bridge_handler.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (aidan-kwon) [reply node/sc/sub_bridge_handler.go]: Of course, `logger.Error`. Thank you for the refleciton.

**Comment by hyunsooda** [comment]
This PR should be delayed until #1248 is finalized.

**Comment by hyunsooda** [comment]
A handler of invalid tx removal is now moved to #1427. The remainder of TODO is to avoid sending duplicated anchoring tx https://github.com/klaytn/klaytn/pull/1295/files#diff-1623298e4e46f32d0a4b35ea8ba2fd0a9de08103caf3820b7ce69293b6e765e5R352-R364.

**Pull request closed without merging**